### PR TITLE
[ISSUE #8515] Optimize some code in InetUtils

### LINKED ISSUE #8515: [Optimize] Some code can be  optimized in InetUtils
`nacos.core.inet.auto-refresh` property is deprecated, isn't it?  so, I think about the `ipAutoRefresh` can be removed and unregistered `IPChangeEvent` .

看了一下代码,发现`nacos.core.inet.auto-refresh`属性已经被弃用,所以我觉得`ipAutoRefresh`这个可以被移除,并且`IPChangeEvent`事件也可以不用注册

image

### PR DESCRIPTION
**Pull request by onewe**
- remove nacos.core.inet.auto-refresh property
- unregistered IPChangeEvent
- add some unit tests for InetUtils

Close #8515

Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [x] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by onewe** [comment]
@KomachiSion 我看代码,这个没有任何调度器来调度.

https://github.com/alibaba/nacos/blob/8a9f6f8bd669ed1c7b41471eae96add61560d5a9/sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java#L77
这里创建了一个 runnable 后面直接就run了

https://github.com/alibaba/nacos/blob/8a9f6f8bd669ed1c7b41471eae96add61560d5a9/sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java#L128

https://github.com/alibaba/nacos/blob/8a9f6f8bd669ed1c7b41471eae96add61560d5a9/sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java#L75
这个时间也没用到

**Comment by onewe** [comment]
这玩意儿的单元测试太难写了

File: sys/src/test/java/com/alibaba/nacos/sys/utils/InetUtilsTest.java [hunk 1]
```diff
@@ -16,14 +16,39 @@
...
+        String selfIP = InetUtils.getSelfIP();
+        Assert.assertTrue(StringUtils.equalsIgnoreCase(selfIP, "1.1.1.1"));
+    
+        System.setProperty(NACOS_SERVER_IP, "1.1.1.2");
+        TimeUnit.MILLISECONDS.sleep(500L);
+    
+        selfIP = InetUtils.getSelfIP();
+        Assert.assertTrue(StringUtils.equalsIgnoreCase(selfIP, "1.1.1.2"));
+        
+        System.clearProperty(NACOS_SERVER_IP);
```
Comment (KomachiSion) [inline sys/src/test/java/com/alibaba/nacos/sys/utils/InetUtilsTest.java:51]: clear operation please move to After method.

File: sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java [hunk 2]
```diff
@@ -156,8 +201,9 @@ public static InetAddress findFirstNonLoopbackAddress() {
                     if (!ignoreInterface(ifc.getDisplayName())) {
                         for (Enumeration<InetAddress> addrs = ifc.getInetAddresses(); addrs.hasMoreElements(); ) {
                             InetAddress address = addrs.nextElement();
-                            boolean isLegalIpVersion = InternetAddressUtil.PREFER_IPV6_ADDRESSES ? address instanceof Inet6Address
-                                    : address instanceof Inet4Address;
+                            boolean isLegalIpVersion =
+                                    InternetAddressUtil.PREFER_IPV6_ADDRESSES ? address instanceof Inet6Address
+                                            : address instanceof Inet4Address;
```
Comment (KomachiSion) [inline sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java:206]: Is reformat by nacos-code-style?

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: sys/src/test/java/com/alibaba/nacos/sys/utils/InetUtilsTest.java (hunk 1 above)
Comment (onewe) [reply sys/src/test/java/com/alibaba/nacos/sys/utils/InetUtilsTest.java:56]: Ok, I have resolved already.

File: sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java (hunk 2 above)
Comment (onewe) [reply sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java:208]: I checked my idea's configuration, I absolutely use the nacos-code-style.

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**